tel2puml/__main__.py

```diff
@@ -18,67 +18,222 @@
+    :type file_path: `str`
+    :return: Config file represented as a dictionary
+    :rtype: `dict`[`str`, `Any`]
+    """
+    with open(file_path, "r") as file:
+        return yaml.safe_load(file)
+
+
+def find_json_files(directory: str) -> list[str]:
+    """Walk a directory path and extract .json files.
+
+    :param directory: The directory path
+    :type directory: `str`
+    :return: List of filepaths
+    :rtype: `list`[`str`]
+    """
+    json_files = []
+
+    for root, _, files in os.walk(directory):
+        for file in files:
+            if file.endswith(".json"):
+                json_files.append(os.path.join(root, file))
+
+    if not json_files:
+        raise FileNotFoundError(
+            f"No .json files were found in directory '{directory}'"
+        )
+
+    return json_files
+
+
+def generate_component_options(
+    command: Literal["otel2puml", "otel2pv", "pv2puml"],
+    args_dict: dict[str, Any],
+) -> tuple[OtelPVOptions | None, PVPumlOptions | None]:
+    """Generate puml options objects based on CLI arguments.
+
+    :param command: The CLI command
+    :type command: `Literal`["otel2puml", "otel2pv", "pv2puml"]
+    :return: A tuple containing component options
+    :rtype: `tuple`[:class:`OtelPVOptions` | `None`, :class:`PVPumlOptions`
+    | `None`]
+    """
+
+    otel_pv_options, pv_puml_options = None, None
+    if command == "otel2puml" or command == "otel2pv":
+        otel_to_pv_obj = OtelToPVArgs(**args_dict)
+        otel_pv_options = OtelPVOptions(
+            config=load_config_from_dict(
+                generate_config(str(otel_to_pv_obj.config_file))
+            ),
+            ingest_data=otel_to_pv_obj.ingest_data,
+        )
+    elif command == "pv2puml":
+        pv_to_puml_obj = PvToPumlArgs(**args_dict)
+        if pv_to_puml_obj.file_paths:
+            file_list = [
+                str(filepath) for filepath in pv_to_puml_obj.file_paths
+            ]
+        elif pv_to_puml_obj.folder_path:
+            file_list = find_json_files(str(pv_to_puml_obj.folder_path))
+        pv_puml_options = PVPumlOptions(
+            file_list=file_list,
+            job_name=pv_to_puml_obj.job_name,
+            group_by_job_id=pv_to_puml_obj.group_by_job,
+        )
+
+    return otel_pv_options, pv_puml_options
+
+
 if __name__ == "__main__":
     args: argparse.Namespace = parser.parse_args()
     args_dict = vars(args)
-    if args_dict["file_paths"]:
-        args_dict.pop("folder_path")
-        args_dict.pop("group_by_job")
-        pv_jobs_from_files_to_puml_file(**args_dict)
-    elif args_dict["group_by_job"]:
-        args_dict.pop("file_paths")
-        pv_events_from_folder_to_puml_file(**args_dict)
-    else:
-        args_dict.pop("file_paths")
-        args_dict.pop("group_by_job")
-        pv_jobs_from_folder_to_puml_file(**args_dict)
+    otel_pv_options, pv_puml_options = generate_component_options(
+        args.command, args_dict
+    )
```

Comment:
will need to run otel2puml here using the options and remember to input the global argument output directory